Library/OcDeviceMiscLib/SetResizableBar.c

```diff
@@ -388,3 +689,178 @@ ResizeGpuBars (
 
   return EFI_NOT_FOUND;
 }
+
+EFI_STATUS
+PciGetNextBusRange (
+  IN OUT EFI_ACPI_ADDRESS_SPACE_DESCRIPTOR  **Descriptors,
+  OUT UINT16                                *MinBus,
+  OUT UINT16                                *MaxBus,
+  OUT BOOLEAN                               *IsEnd
+  )
+{
+  *IsEnd = FALSE;
+
+  //
+  // When *Descriptors is NULL, Configuration() is not implemented, so assume
+  // range is 0~PCI_MAX_BUS
+  //
+  if ((*Descriptors) == NULL) {
+    *MinBus = 0;
+    *MaxBus = PCI_MAX_BUS;
+    return EFI_SUCCESS;
+  }
+
+  //
+  // *Descriptors points to one or more address space descriptors, which
+  // ends with a end tagged descriptor. Examine each of the descriptors,
+  // if a bus typed one is found and its bus range covers bus, this handle
+  // is the handle we are looking for.
+  //
+
+  while ((*Descriptors)->Desc != ACPI_END_TAG_DESCRIPTOR) {
+    if ((*Descriptors)->ResType == ACPI_ADDRESS_SPACE_TYPE_BUS) {
+      *MinBus = (UINT16)(*Descriptors)->AddrRangeMin;
+      *MaxBus = (UINT16)(*Descriptors)->AddrRangeMax;
+      (*Descriptors)++;
+      return (EFI_SUCCESS);
+    }
+
+    (*Descriptors)++;
+  }
+
+  if ((*Descriptors)->Desc == ACPI_END_TAG_DESCRIPTOR) {
+    *IsEnd = TRUE;
+  }
+
+  return EFI_SUCCESS;
```

Comment:
Why return a status when the function can only ever succeed? Maybe return IsEnd?